model/src/pyrenew/model/admissionsmodel.py

```diff
@@ -250,28 +251,35 @@ def sample(
             infections=basic_model.latent_infections,
             **kwargs,
         )
-
-        # Sampling the hospital admissions
-        if self.observation_process is not None:
-            if (observed_admissions is not None) and (padding > 0):
-                sampled_na = jnp.repeat(jnp.nan, padding)
-
-                sampled_observed, *_ = self.sample_admissions_process(
-                    predicted=latent[padding:],
-                    observed_admissions=observed_admissions[padding:],
+        i0_size = len(latent) - n_timepoints
+        if self.observation_process is None:
+            sampled = None
+        else:
+            if observed_admissions is None:
+                sampled_obs, *_ = self.sample_admissions_process(
+                    predicted=latent,
+                    observed_admissions=observed_admissions,
                     **kwargs,
                 )
-
-                sampled = jnp.hstack([sampled_na, sampled_observed])
-
             else:
-                sampled, *_ = self.sample_admissions_process(
-                    predicted=latent,
-                    observed_admissions=observed_admissions,
+                observed_admissions = au.pad_x_to_match_y(
+                    observed_admissions, latent, jnp.nan, pad_direction="start"
+                )
+
+                sampled_obs, *_ = self.sample_admissions_process(
+                    predicted=latent[i0_size + padding :],
+                    observed_admissions=observed_admissions[
+                        i0_size + padding :
+                    ],
                     **kwargs,
                 )
-        else:
-            sampled = None
+            # this is to accommodate the current version of test_model_hosp_no_obs_model. Not sure if we want this behavior
```

Comment:
Just calling attention to this comment

Reply:
In general seeing this many switches suggests to me that this can and should be refactored and modularized. But separate PR I think. Let me know if you disagree.